raiden/api/rest.py

```diff
@@ -273,15 +276,19 @@ def run(self, host='127.0.0.1', port=5001, **kwargs):
         self.flask_app.run(host=host, port=port, **kwargs)
 
     def start(self, host='127.0.0.1', port=5001):
-        # WSGI expects a stdlib logger, with structlog there's conflict of method names
-        wsgi_log = logging.getLogger(__name__ + '.pywsgi')
-        self.wsgiserver = WSGIServer(
-            (host, port),
-            self.flask_app,
-            log=wsgi_log,
-            error_log=wsgi_log,
-        )
-        self.wsgiserver.start()
+        try:
+            # WSGI expects a stdlib logger, with structlog there's conflict of method names
+            wsgi_log = logging.getLogger(__name__ + '.pywsgi')
+            self.wsgiserver = WSGIServer(
+                (host, port),
+                self.flask_app,
+                log=wsgi_log,
+                error_log=wsgi_log,
+            )
+            self.wsgiserver.start()
+        except socket.error as e:
+            if e.errno == errno.EADDRINUSE:
+                raise APIServerPortInUseError()
```

Comment:
There needs to be a `raise` here for the non `EADDRINUSE` case, otherwise we're swallowing exceptions.